src/object/cli_obj.c

```diff
@@ -688,24 +689,127 @@ obj_rw_req_reassemb(struct dc_object *obj, daos_obj_rw_t *args,
 	return rc;
 }
 
+static bool
+obj_ec_tgt_in_err(uint32_t *err_list, uint32_t nerrs, uint16_t tgt_idx)
+{
+	uint32_t	i;
+
+	for (i = 0; i < nerrs; i++) {
+		if (err_list[i] == tgt_idx)
+			return true;
+	}
+	return false;
+}
+
+static int
+obj_ec_get_degrade(struct obj_auxi_args *obj_auxi, uint16_t fail_tgt_idx,
+		   uint32_t *parity_tgt_idx)
+{
+	struct obj_reasb_req	*reasb_req = &obj_auxi->reasb_req;
+	uint16_t		 p = obj_ec_parity_tgt_nr(reasb_req->orr_oca);
+	uint16_t		 k = obj_ec_data_tgt_nr(reasb_req->orr_oca);
+	uint32_t		*err_list;
+	uint32_t		 nerrs, i;
+	bool			 with_parity = false;
+
+	D_ASSERT(fail_tgt_idx < k + p);
```

Comment:
should be "k" or "k+p", I assume call into this only if failt_tgt_idx < k?

Reply:
it possibly with fail_tgt_idx > k, for example a 4+2 EC obj, firstly found that shard 3 (last data shard) failed, and later try to recovery previous full-stripes. In the recovery task, it possibly will find that parity target shard 4 failed again/also, in that case it need to init another set of recovery task. (In PR2886).